cmake/protobuf-generate-cpp.cmake.in

```diff
@@ -64,6 +64,61 @@ function(PROTOBUF_CATKIN_GENERATE_CPP SRCS HDRS)
   set(${HDRS} ${${HDRS}} PARENT_SCOPE)
 endfunction()
 
+function(PROTOBUF_CATKIN_GENERATE_JAVA SRCS)
+  if(NOT ARGN)
+    message(SEND_ERROR "Error: PROTOBUF_CATKIN_GENERATE_JAVA() called without any proto files")
+    return()
+  else()
+    message(STATUS "FILES: ${ARGN}")
+  endif(NOT ARGN)
+
+  if(PROTOBUF_GENERATE_JAVA_APPEND_PATH)
+    set(_protobuf_include_path)
+    # Create an include path for each file specified
+    foreach(FIL ${ARGN})
+      get_filename_component(ABS_FIL ${FIL} ABSOLUTE)
+      get_filename_component(ABS_PATH ${ABS_FIL} PATH)
+      list(FIND _protobuf_include_path ${ABS_PATH} _contains_already)
+      if(${_contains_already} EQUAL -1)
+          list(APPEND _protobuf_include_path -I ${ABS_PATH})
+      endif()
+    endforeach()
+  else()
+    set(_protobuf_include_path -I ${CMAKE_CURRENT_SOURCE_DIR})
+  endif()
```

Comment:
You could make this a separate function to avoid code duplication for the C++ and Java parts.